doc/architectural/compilation-and-development.md

```diff
@@ -302,3 +302,22 @@ each function; and the "call graph", which includes context, and tells you how
 much time was spent within each call stack. For more information see online
 tutorials, like
 https://ftp.gnu.org/old-gnu/Manuals/gprof-2.9.1/html_chapter/gprof_5.html
+
+\section compilation-and-development-section-max-performance Maximum Performance
+
+On Intel with gcc-compatible compilers (i.e. gcc, clang, icc) you may
+use the `-Denable_marchnative=1` option, to pass `-march=native` to cc
+and c++.
+
+    CXXFLAGS="-O3" CFLAGS="-O3" cmake -S . -Bbuild-native -Denable_marchnative=1
+    make -j4 -C build-native
```

Comment:
I think this would be clearer if it only showed the new and important flags, we don't need to show the specification of the `-O3` flag or `-j4`. So something elike:
```
cmake -S . -Bbuild-native -Denable_marchnative=1
make -C build-native
```